### DISCUSSION TITLE
posts/add-comments-section-to-your-astro-blog

### DISCUSSION THREAD

**Discussion by giscus** [yjose/elazizi.com#117] [Blog posts]
# posts/add-comments-section-to-your-astro-blog

A quick guide on how to add comments section to your astro blog

https://elazizi.com/posts/add-comments-section-to-your-astro-blog

**Comment by thomasledoux1** [comment]
Great idea! I did the same on my blog, also using Astro and Giscus 👌

**Comment by sergey-stukov** [comment]
test

**Reply by yjose** [reply]
It's working as expected 😂 thank your for the test @sergey-stukov

**Comment by pthurmond-vmlyr** [comment]
Thanks, I just used this to add comments to my website, https://www.patrickthurmond.com/blog/

It was super easy to implement. I just had to change the Tailwind styles a bit and it looks great. Also, I mapped mine based on the pathname so that it would work great in local testing too. I'm sure the other mappings would probably be fine too.

Thanks for this!

**Comment by ryfazrin** [comment]
Thanks! This is great Idea. I will use it for my blog 👌

**Comment by jingxiangmo** [comment]
Awesome! Thank you

**Comment by jvmonjo** [comment]
Excellent post! Thanks. I’m using Astropaper also as my blog theme 👌

**Comment by xkeshav** [comment]
wow, works like charm. Thanks. 

there should be one correction in the code snippet

```js
import PostComments from "@components/Comments.astro";
```

**Reply by yjose** [reply]
fixed, thank you

**Comment by armno** [comment]
Thanks a lot! My blog has comments now.

**Comment by jpasholk** [comment]
I’m going to try this tonight or tomorrow!

**Comment by rajeevmyname** [comment]
Excellent post...I will use this on my blog too

**Comment by herijaya** [comment]
josss

**Comment by tombohub** [comment]
Can you hear me?

**Comment by baoduy** [comment]
this is amazing

**Comment by joeattardi** [comment]
Awesome! Just what I was looking for. Thanks!

**Comment by ITZSHOAIB** [comment]
Thanks for a detailed blog on this topic. My blog [https://4techloverz.com](https://4techloverz.com) is now also having Giscus Comments integrated. 

Although I struggled a bit to configure dual theme, but I eventually achieved it using [`sendMessage`](https://github.com/giscus/giscus/blob/main/ADVANCED-USAGE.md#parent-to-giscus-message-events) method on theme change.

**Comment by TotallyInformation** [comment]
This is a great tool, many thanks for explaining it. However, it does somewhat lack on the efficiency front.

For example, it would be great if you added minimised JS and CSS - perhaps using ESBUILD which is pretty simple to use.

**Reply by yjose** [reply]
I'm not sure what you mean here, but normally, the Giscus client script is sourced from the CDN. How can we optimize it?

**Reply by TotallyInformation** [reply]
If you look at my new blog at https://www.totallyinformation.net which is delivered using Astro on the Cloudflare Pages platform, it should be lightning quick to load. But with Giscus present, it is taking some 1.5 seconds which is massive. If you check the networks tab in the browser dev tools and do a ctrl-F5 to reload without the cache, you will see why it is so slow. A very large chunk of the loads are due to Giscus.

![image](https://github.com/user-attachments/assets/15a8f657-1c06-4611-8ab8-678af3da144e)

A closer examination of the performance shows that Giscus is the culprit. There are 11 separate JavaScript files loaded. 3 separate CSS files. These are despite the fact that you are clearly using a bundler.

**Comment by ivomynttinen** [comment]
Pretty cool! Just implemented it on my own [Blog](https://ivomynttinen.com/blog/)!

I noticed that your version does look a bit poor when you view in light mode using the toggle in the header. Since I had the same problem, I've [extended](https://github.com/ivomynttinen/ivo-mynttinen/blob/main/src/components/Comments.astro) the implementation by using 2 custom themes (dark and light) and then reload the script when using the theme toggle. Feels kinda dirty, but couldn't really find another way to use custom css variables and support both light and dark mode...

**Comment by jeminpro** [comment]
Well written post, I will try it out on my website

**Comment by nullx5** [comment]
I did not know him, but he is interesting, he wise of utterances is similar but uses issues. ` [>_] `🤪

**Comment by jkfids** [comment]
Is it possible to have reactions only and disable comments? Or is there a better solution?

**Comment by Rednegniw** [comment]
Great tutorial, thank you!

**Comment by jayjarosz** [comment]
test

**Comment by RubenHeeren** [comment]
This is a test

**Comment by zakarialaoui10** [comment]
Wordpress version :
- Plugin : https://wordpress.org/plugins/wiscus/
- Repo : https://github.com/zakarialaoui10/wiscus
